include/onnxruntime/interface/framework/kernel.h

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "interface/common/data_types.h"
+#include "interface/framework/tensor.h"
+#include "core/common/status.h"
+
+namespace onnxruntime {
+
+namespace interface {
+
+using TensorShape = std::vector<int64_t>;
+
+struct IKernelContext {
+  virtual ~IKernelContext() = default;
+  virtual const void* InputData(int index) const = 0;
+  virtual const int64_t* InputShape(int index, size_t* num_dims) const = 0;
+  virtual void* AllocateOutput(int index, const TensorShape& shape) = 0;
+};
+
+struct IArg {
+  virtual ~IArg() = default;
+};
+
+using ArgPtr = std::unique_ptr<IArg>;
+using ArgPtrs = std::vector<ArgPtr>;
+
+template <typename T>
+struct ITensor : public IArg {
```

Comment:
Are you going to move this class to interface/framework/Tensor.h?

Shall we consolidate this tensor interface with the one in interface/graph/graph.h ?